src/Integration/Notifications/SonarQubeNotifications.cs

```diff
@@ -0,0 +1,211 @@
+using System.ComponentModel.Composition;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Timers;
+using Microsoft.VisualStudio.Shell;
+using SonarLint.VisualStudio.Integration.Service;
+using SonarLint.VisualStudio.Integration.State;
+using SonarLint.VisualStudio.Integration.WPF;
+
+using CancellationTokenSource = System.Threading.CancellationTokenSource;
+
+namespace SonarLint.VisualStudio.Integration.Notifications
+{
+    [Export(typeof(ISonarQubeNotifications))]
+    internal class SonarQubeNotifications : ViewModelBase, ISonarQubeNotifications
+    {
+        private string text = "You have no events.";
+        private bool hasUnreadEvents;
+        private bool isVisible;
+        private bool isBalloonTooltipVisible;
+
+        private DateTimeOffset lastRequestDate = DateTimeOffset.MinValue;
+        private readonly ITimer timer;
+        private readonly IStateManager stateManager;
+        private readonly ISonarQubeServiceWrapper sonarQubeService;
+        private readonly CancellationTokenSource cancellation = new CancellationTokenSource();
+        private bool isEnabled = true;
+
+        [ImportingConstructor]
+        [ExcludeFromCodeCoverage] // Do not unit test MEF constructor
+        internal SonarQubeNotifications(IHost host)
+            : this(host.SonarQubeService, host.VisualStateManager,
+                  new TimerWrapper { Interval = 10000 /* should be 60sec */ })
+        {
+        }
+
+        internal SonarQubeNotifications(ISonarQubeServiceWrapper sonarQubeService,
+            IStateManager stateManager, ITimer timer)
+        {
+            this.timer = timer;
+            this.sonarQubeService = sonarQubeService;
+            this.stateManager = stateManager;
+
+            timer.Elapsed += OnTimerElapsed;
+            NotificationEvents = new ObservableCollection<NotificationEvent>();
+        }
+
+        public string Text
+        {
+            get
+            {
+                return text;
+            }
+            set
+            {
+                SetAndRaisePropertyChanged(ref text, value);
+            }
+        }
+
+        public bool HasUnreadEvents
+        {
+            get
+            {
+                return hasUnreadEvents;
+            }
+
+            set
+            {
+                SetAndRaisePropertyChanged(ref hasUnreadEvents, value);
+
+                UpdateTooltipText();
+            }
+        }
+
+        public bool IsVisible
+        {
+            get
+            {
+                return isVisible;
+            }
+            set
+            {
+                SetAndRaisePropertyChanged(ref isVisible, value);
+            }
+        }
+
+        public bool IsEnabled
+        {
+            get
+            {
+                return isEnabled;
+            }
+            set
+            {
+                SetAndRaisePropertyChanged(ref isEnabled, value);
+            }
+        }
+
+        public bool IsBalloonTooltipVisible
```

Comment:
Convention is to use `ToolTip` and not `Tooltip`.

Reply:
Agree, but let's use BalloonTip. The property name was inspired from [here](https://msdn.microsoft.com/en-us/library/ms160064(v=vs.110).aspx).